mep/people/forms.py

```diff
@@ -110,22 +114,39 @@ class MemberSearchForm(FacetForm):
     }))
     membership_dates = RangeField(label='Membership Dates', required=False,
         widget=RangeWidget(attrs={'size': 4}))
+    birth_year = RangeField(required=False,
+        widget=RangeWidget(attrs={'size': 4}))
 
-    def set_membership_dates_placeholder(self, min_year, max_year):
-        '''Set the min, max, and placeholder values for
-        :class:`mep.common.forms.RangeWidget` associated with membership_dates.'''
-
-        start_widget, end_widget = \
-            self.fields['membership_dates'].widget.widgets
-
-        # set placeholders for widgets individually
-        start_widget.attrs['placeholder'] = min_year
-        end_widget.attrs['placeholder'] = max_year
-        # valid min and max for both via multiwidget
-        self.fields['membership_dates'].widget.attrs.update({
-            'min': min_year,
-            'max': max_year
-        })
+    def set_daterange_placeholders(self, min_max_conf):
+        '''Set the min, max, and placeholder values for all fields associated
+        with :class:`mep.common.forms.RangeWidget`.
+
+        :param min_max_conf: set of (min, max) tuples
+        :type min_max_conf: dict
+        :rtype: None
+        '''
+        for field, field_obj in self.fields.items():
+            if isinstance(field_obj, RangeField):
+                # Lookup up alias in mapping of range fields to solr fields
+                # or look for field in dict
+                stats_field_name = self.range_field_map.get(field, field)
+                min_max = min_max_conf.get(stats_field_name, None)
+
+                # Continue if a field isn't found for whatever reason
+                # so that other facets can be processed correctly.
+                if min_max is None:
+                    continue
+
+                min_year, max_year = min_max
```

Comment:
What do you think about moving this logic to a new method on either `RangeField` or `RangeWidget` ?

Reply:
That works and fixes the cognitive complexity complaints. I'll do just that, thanks for the suggestion.